<a href="https://colab.research.google.com/github/IsagiYoichi640/Implementation-of-Linear-Regression/blob/main/Implementation_of_Linear_Regression.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from google.colab import files
file=files.upload()

Saving petrol_consumption.csv to petrol_consumption.csv


In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as no
%matplotlib inline

In [4]:
ds=pd.read_csv("/content/petrol_consumption.csv")

In [5]:
ds.head()

Petrol_tax  Average_income  Paved_Highways  Population_Driver_licence(%)  \
0         9.0            3571            1976                         0.525   
1         9.0            4092            1250                         0.572   
2         9.0            3865            1586                         0.580   
3         7.5            4870            2351                         0.529   
4         8.0            4399             431                         0.544   

   Petrol_Consumption  
0                 541  
1                 524  
2                 561  
3                 414  
4                 410

In [6]:
ds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48 entries, 0 to 47
Data columns (total 5 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Petrol_tax                    48 non-null     float64
 1   Average_income                48 non-null     int64  
 2   Paved_Highways                48 non-null     int64  
 3   Population_Driver_licence(%)  48 non-null     float64
 4   Petrol_Consumption            48 non-null     int64  
dtypes: float64(2), int64(3)
memory usage: 2.0 KB


#Findings
* There are 5 numerical columns and 0 descriptive columns
* There are 48 entries

In [7]:
ds.isnull().sum()

Petrol_tax                      0
Average_income                  0
Paved_Highways                  0
Population_Driver_licence(%)    0
Petrol_Consumption              0
dtype: int64

#findings
* there are no null values

In [8]:
#dividing data
X=ds[["Petrol_tax","Average_income","Paved_Highways","Population_Driver_licence(%)"]]
y=ds[["Petrol_Consumption"]]

In [9]:
#splitting data into testing and training
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=0)

In [10]:
#build the algorithm
from sklearn.linear_model import LinearRegression
rg=LinearRegression()

In [11]:
#train the model on the dataset
rg.fit(X_train,y_train)

LinearRegression()

In [24]:
#make predictions
y_pred=rg.predict(X_test)

In [44]:
#dataframe-Predicted values vs actual values
df=pd.DataFrame({'Actual Values':y_test.values.flatten(),'Predicted Values':y_pred.flatten()})
display(df)

Actual Values  Predicted Values
0            534        469.391989
1            410        545.645464
2            577        589.668394
3            571        569.730413
4            577        649.774809
5            704        646.631164
6            487        511.608148
7            587        672.475177
8            467        502.074782
9            580        501.270734

In [45]:
#measure the accuracy of the algorithm
from sklearn import metrics
print("mean absoulute error:", metrics.mean_absolute_error(y_test,y_pred))

mean absoulute error: 56.822247478964734


In [46]:
print("r2 score:",metrics.r2_score(y_test,y_pred))

r2 score: 0.2036193241012163


#Fidings
* Mean Absolute error: Tells us the difference between the predicted values and the actual values.


In [47]:
#accuracy of model
print("Accuracy of model:", rg.score(X,y)*100)

Accuracy of model: 67.37990827579196


#fidings
* Linear regression accurately predicted petrol consumption by 67%.